### PR TITLE
feat(crux-ui): use next link

### PR DESCRIPTION
**Pull request by bvektor**

### CODE REVIEW

**Comment by polaroi8d** [comment]
E2E Tests are failing 🥳

File: web/crux-ui/src/components/nodes/dyo-node-card.tsx [hunk 1]
```diff
@@ -6,32 +6,45 @@ import { DyoLabel } from '@app/elements/dyo-label'
...
-        </DyoHeading>
-
-        {!hideConnectionInfo ? <NodeStatusIndicator status={node.status} /> : null}
-      </div>
+      {typeof onNameClick === 'string' ? (
+        <Link href={onNameClick}>
+          <a className="flex flex-row">{title}</a>
+        </Link>
+      ) : (
+        <div className={clsx(onNameClick ? 'cursor-pointer' : null, 'flex flex-row')} onClick={onNameClick}>
```
Comment (m8vago) [inline web/crux-ui/src/components/nodes/dyo-node-card.tsx:44]: why do we need this case?

File: web/crux-ui/src/components/notifications/notification-card.tsx [hunk 2]
```diff
@@ -5,32 +5,44 @@ import { NotificationDetails } from '@app/models'
...
-
-        <DyoHeading className="text-xl text-bright ml-2 my-auto mr-auto truncate" element="h3" onClick={onClick}>
-          {notification.name}
-        </DyoHeading>
-      </div>
+      {titleHref ? (
+        <Link href={titleHref}>
+          <a className="flex flex-row">{title}</a>
+        </Link>
+      ) : (
```
Comment (m8vago) [inline web/crux-ui/src/components/notifications/notification-card.tsx:43]: why do we need this case?

File: web/crux-ui/src/components/products/product-card.tsx [hunk 3]
```diff
@@ -5,34 +5,49 @@ import { utcDateToLocale } from '@app/utils'
...
-            alt={t('altPicture', { name: product.name })}
-            width={100}
-            height={100}
-            onClick={onClick}
-          />
+          {imageWrapper()}
 
           <div className="flex flex-col flex-grow">
-            <DyoHeading element="h5" className="text-lg text-bright ml-4" onClick={onClick}>
+            <DyoHeading element="h5" className="text-lg text-bright ml-4" onClick={onClick} href={href}>
```
Comment (m8vago) [inline web/crux-ui/src/components/products/product-card.tsx:50]: is the onclick ever used?

File: web/crux-ui/src/elements/dyo-anchor-action.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  anchors: AnchorActions
+  children: React.ReactNode
+  disabled?: boolean
+}
+
+const AnchorAction = (props: AnchorActionProps) => {
+  const { anchors, href: propsHref, children, disabled, ...forwardedProps } = props
+
+  // eslint-disable-next-line no-console
+  console.assert(Object.keys(anchors).includes(propsHref), 'AnchorAction not found for', propsHref)
```
Comment (polaroi8d) [inline web/crux-ui/src/elements/dyo-anchor-action.tsx:21]: Are we need this?

File: web/crux-ui/src/components/registries/registry-card.tsx [hunk 5]
```diff
@@ -7,31 +7,45 @@ import { Registry } from '@app/models'
...
-
-        <DyoHeading className="text-xl text-bright font-semibold ml-2 my-auto mr-auto" element="h3">
-          {registry.name}
-        </DyoHeading>
-      </div>
+      {titleHref ? (
+        <Link href={titleHref}>
+          <a className="flex flex-row">{title}</a>
+        </Link>
+      ) : (
```
Comment (m8vago) [inline web/crux-ui/src/components/registries/registry-card.tsx:44]: same question

File: web/crux-ui/src/hooks/use-anchor-actions.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    const action = actions[anchor]
+    if (!action) {
+      return
+    }
+
+    const result = action()
+    if (typeof result === 'object') {
+      triggered.current = anchor
```
Comment (m8vago) [inline web/crux-ui/src/hooks/use-anchor-actions.ts:32]: i think we should move this outside of the if statement

File: web/crux-ui/src/pages/products/[productId]/versions/[versionId]/deployments/[deploymentId].tsx [hunk 7]
```diff
@@ -58,6 +60,19 @@ const DeploymentDetailsPage = (props: DeploymentDetailsPageProps) => {
...
+    const url = await actions.onCopyDeployment()
+    if (!url) {
+      return
+    }
+
+    router.push(url)
+  }
+
+  const anchors = useAnchorActions({
+    copyDeployment: () => onCopyDeployment(),
```
Comment (m8vago) [inline web/crux-ui/src/pages/products/[productId]/versions/[versionId]/deployments/[deploymentId].tsx:73]: this should be just copyDeployment: onCopyDeployment. or even better: just rename the onCopyDeployment function

File: web/crux-ui/src/elements/dyo-anchor-action.tsx (hunk 4 above)
Comment (bvektor) [reply web/crux-ui/src/elements/dyo-anchor-action.tsx:21]: After discussing it with @m8vago we came to the conclusion: _yes_

**Review by polaroi8d — APPROVED** [review]
LGTM (informal)

File: web/crux-ui/src/components/products/product-card.tsx [hunk 8]
```diff
@@ -5,34 +5,38 @@ import { utcDateToLocale } from '@app/utils'
...
 import useTranslation from 'next-translate/useTranslation'
 import Image from 'next/image'
+import Link from 'next/link'
 import ProductTypeTag from './product-type-tag'
 
 interface ProductCardProps {
   className?: string
   product: Product
-  onClick?: () => void
+  href?: string
```
Comment (m8vago) [inline web/crux-ui/src/components/products/product-card.tsx:14]: let's call this one tilteHref as well

File: web/crux-ui/src/components/products/versions/version-view-list.tsx [hunk 9]
```diff
@@ -85,13 +82,11 @@ const VersionViewList = (props: VersionViewListProps) => {
...
       </div>
       <div className="inline-block">
-        <Image
-          className="cursor-pointer"
-          src="/settings.svg"
-          width={24}
-          height={24}
-          onClick={() => onImageSettings(item)}
-        />
+        <Link href={imageConfigUrl(state.productId, state.versionId, item.id)}>
```
Comment (m8vago) [inline web/crux-ui/src/components/products/versions/version-view-list.tsx:85]: this needs passHref i think

Comment (bvektor) [reply web/crux-ui/src/components/products/versions/version-view-list.tsx:85]: Afaik no, passHRef is only needed if the child is a custom component that wraps an <a> tag, but we have the <a> as a direct child, so we don't need to force it.

**Review by m8vago — APPROVED** [review]

**Review by polaroi8d — APPROVED** [review]
Again/

**Pull request merged**